tools/topology/topology1/sof-glk-es8336.m4

```diff
@@ -0,0 +1,194 @@
+')
+
+
+# Low Latency playback pipeline 1 on PCM 0 using max 2 channels of s32le.
+# 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+	1, 0, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
+# Low Latency capture pipeline 2 on PCM 0 using max 2 channels of s32le.
+# 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-low-latency-capture.m4,
+	2, 0, 2, s32le,
+	1000, 0, 0,
+	48000, 48000, 48000)
+
+ifelse(HDMI, `1',
+`
+# Low Latency playback pipeline 5 on PCM 5 using max 2 channels of s32le.
+# 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+        5, 5, 2, s32le,
+        1000, 0, 0,
+	48000, 48000, 48000)
+
+# Low Latency playback pipeline 6 on PCM 6 using max 2 channels of s32le.
+# 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+        6, 6, 2, s32le,
+        1000, 0, 0,
+	48000, 48000, 48000)
+
+# Low Latency playback pipeline 7 on PCM 7 using max 2 channels of s32le.
+# 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+        7, 7, 2, s32le,
+        1000, 0, 0,
+	48000, 48000, 48000)
+'
+)
+
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     deadline, priority, core, time_domain)
+
+# playback DAI is SSP2 using 2 periods
+# Buffers use s24le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+	1, SSP, SSP_NUM, `SSP'SSP_NUM`-Codec',
+	PIPELINE_SOURCE_1, 2, s24le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+# capture DAI is SSP2 using 2 periods
+# Buffers use s24le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	2, SSP, SSP_NUM, `SSP'SSP_NUM`-Codec',
+	PIPELINE_SINK_2, 2, s24le,
+	1000, 0, 0, SCHEDULE_TIME_DOMAIN_DMA)
+
+ifelse(HDMI, `1',
+`
+# playback DAI is iDisp1 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+        5, HDA, 3, iDisp1,
+        PIPELINE_SOURCE_5, 2, s32le,
+        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp2 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+        6, HDA, 4, iDisp2,
+        PIPELINE_SOURCE_6, 2, s32le,
+        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+
+# playback DAI is iDisp3 using 2 periods
+# Buffers use s32le format, 1000us deadline on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-playback.m4,
+        7, HDA, 5, iDisp3,
+        PIPELINE_SOURCE_7, 2, s32le,
+        1000, 0, 0, SCHEDULE_TIME_DOMAIN_TIMER)
+'
+)
+
+PCM_DUPLEX_ADD(ES8336, 0, PIPELINE_PCM_1, PIPELINE_PCM_2)
+
+ifelse(HDMI, `1',
+`
+PCM_PLAYBACK_ADD(HDMI 1, 5, PIPELINE_PCM_5)
+PCM_PLAYBACK_ADD(HDMI 2, 6, PIPELINE_PCM_6)
+PCM_PLAYBACK_ADD(HDMI 3, 7, PIPELINE_PCM_7)
```

Comment:
This won't work as HDMI is a variable, it gets replaced and you won't get "HDMI 1" in the output:

```
SectionPCM."1 1" {
        dai."1 1 5" {
                id "5"
        }
```

Reply:
thanks @kv2019i for finding this out.